### PR TITLE
Issue 204

### LINKED ISSUE #204: Use latest versions of source data
We always download version 27 for ChEMBL, but we should change this to using the latest version

### PR DESCRIPTION
**Pull request by jsstevenson**
Close #204 

Also changed PROJECT_ROOT to APP_ROOT

### CODE REVIEW

File: therapy/etl/chembl.py [hunk 1]
```diff
@@ -16,46 +16,34 @@
...
 
-    def _extract_data(self, *args, **kwargs):
+    def _extract_data(self):
         """Extract data from the ChEMBL source."""
-        logger.info('Extracting chembl_27.db...')
-        if 'data_path' in kwargs:
-            chembl_db = kwargs['data_path']
+        self._src_data_dir.mkdir(exist_ok=True, parents=True)
+        chembl_files = list(self._src_data_dir.glob('chembl_*.db'))
+        if chembl_files:
```
Comment (korikuzma) [inline therapy/etl/chembl.py:23]: This does not get the latest chembl version if previous versions are already present

**Review by korikuzma — CHANGES_REQUESTED** [review]

File: therapy/etl/chembl.py (hunk 1 above)
Comment (jsstevenson) [reply therapy/etl/chembl.py:23]: Yeah, is that intended behavior? I was thinking about doing an across-the-board `--force-update` option to fetch the latest version of everything if we want to make that available.

Comment (korikuzma) [reply therapy/etl/chembl.py:23]: Yes, that was the reason for this issue. I think it makes sense for the default behavior to fetch the latest version. If they didn't want to update a source, they would just exclude it from the `normalizer` flag in CLI

Comment (jsstevenson) [reply therapy/etl/chembl.py:23]: Hm... for reproduceability purposes, we may want to be able to load from older files somehow, but maybe we can cross that bridge later.

Comment (korikuzma) [reply therapy/etl/chembl.py:23]: Ya we have this old [issue](https://github.com/cancervariants/gene-normalization/issues/15) in gene-normalizer regarding this

File: therapy/etl/chembl.py [hunk 2]
```diff
@@ -16,46 +16,34 @@
...
-        """Transform SQLite data to JSON."""
+    def _download_data(self):
+        """Download latest ChEMBL database file from EBI."""
+        os.environ['PYSTOW_HOME'] = str(self._src_data_dir.parent.absolute())
+        logger.info('Downloading latest ChEMBL version...')
+        tmp_path = chembl_downloader.download_extract_sqlite()
+        logger.info('ChEMBL download complete.')
+        self._version = chembl_downloader.latest()
+        shutil.move(tmp_path, self._src_data_dir)
+        shutil.rmtree(tmp_path.parent.parent.parent)
```
Comment (korikuzma) [inline therapy/etl/chembl.py:43]: Should `bioversions` and `bioregistry` directories also be removed? They're both empty

File: therapy/etl/hemonc.py [hunk 3]
```diff
@@ -17,7 +17,7 @@
 class HemOnc(Base):
     """Docstring"""
 
-    def __init__(self, database, data_path: Path = PROJECT_ROOT / 'data'):
+    def __init__(self, database, data_path: Path = APP_ROOT / 'data'):
```
[the remark this answers is not in the document]
Comment (jsstevenson) [reply therapy/etl/hemonc.py:20]: I think we either have to provide a default in __init__ or pass it explicitly when the CLI creates each instance. But we can at least have them all reference a common variable to make consistency easier.

File: therapy/etl/chembl.py (hunk 2 above)
Comment (jsstevenson) [reply therapy/etl/chembl.py:43]: Ooh maybe. I thought they might get used for caching the version numbers during runtime but not totally clear, I'll see what happens if you clear them out

**Pull request closed without merging**